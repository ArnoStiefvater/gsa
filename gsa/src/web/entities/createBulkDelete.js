/* Copyright (C) 2020 Greenbone Networks GmbH
 *
 * SPDX-License-Identifier: AGPL-3.0-or-later
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Affero General Public License
 * as published by the Free Software Foundation, either version 3
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */

import SelectionType from 'web/utils/selectiontype';

const createBulkDelete = ({
  entities,
  filteredEntities,
  selected,
  selectionType,
  deleteFunc,
  refetch,
  onError,
}) => {
  const handleBulkDelete = () => {
    const toDelete = [];

    if (selectionType === SelectionType.SELECTION_USER) {
      selected.forEach(item => {
        const promise = deleteFunc(item.id);

        toDelete.push(promise);
      });
    } else if (selectionType === SelectionType.SELECTION_PAGE_CONTENTS) {
      entities.forEach(entity => {
        const promise = deleteFunc(entity.id);

        toDelete.push(promise);
      });
    } else {
      filteredEntities.forEach(entity => {
        const promise = deleteFunc(entity.id);
        toDelete.push(promise);
      });
    }

    return Promise.all([...toDelete]).then(refetch, onError);
  };

  return handleBulkDelete;
};

export default createBulkDelete;
