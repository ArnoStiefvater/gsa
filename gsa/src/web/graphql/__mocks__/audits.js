/* Copyright (C) 2021 Greenbone Networks GmbH
 *
 * SPDX-License-Identifier: AGPL-3.0-or-later
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Affero General Public License
 * as published by the Free Software Foundation, either version 3
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */

import {
  createGenericQueryMock,
  createGenericMutationResult,
} from 'web/utils/testing';
import {
  CREATE_AUDIT,
  MODIFY_AUDIT,
  RESUME_AUDIT,
  START_AUDIT,
  STOP_AUDIT,
} from '../audits';

export const createAuditInput = {
  name: 'a1',
  comment: 'bar',
  policyId: 'p1',
  scannerId: 's1',
  targetId: 't1',
};

const createAuditResult = {
  createAlert: {
    id: '657',
    status: 200,
  },
};

export const createCreateAuditQueryMock = () =>
  createGenericQueryMock(CREATE_AUDIT, createAuditResult, {
    input: createAuditInput,
  });

export const modifyAuditInput = {
  id: '657',
  name: 'a1',
  comment: 'bar',
  policyId: 'p1',
  scannerId: 's1',
  targetId: 't1',
};

const modifyAuditResult = {
  modifyAudit: {
    ok: true,
  },
};

export const createModifyAuditQueryMock = () =>
  createGenericQueryMock(MODIFY_AUDIT, modifyAuditResult, {
    input: modifyAuditInput,
  });

const stopAuditResult = createGenericMutationResult('stopAudit');

export const createStopAuditQueryMock = auditId =>
  createGenericQueryMock(STOP_AUDIT, stopAuditResult, {id: auditId});

const resumeAuditResult = createGenericMutationResult('resumeAudit');

export const createResumeAuditQueryMock = auditId =>
  createGenericQueryMock(RESUME_AUDIT, resumeAuditResult, {id: auditId});

export const createStartAuditQueryMock = (auditId, reportId) => {
  const startAuditResult = {
    startAudit: {
      reportId,
    },
  };

  return createGenericQueryMock(START_AUDIT, startAuditResult, {id: auditId});
};
