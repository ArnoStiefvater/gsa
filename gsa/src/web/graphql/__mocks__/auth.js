/* Copyright (C) 2020 Greenbone Networks GmbH
 *
 * SPDX-License-Identifier: AGPL-3.0-or-later
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Affero General Public License
 * as published by the Free Software Foundation, either version 3
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */

import {GET_CURRENT_USER_IS_AUTHENTICATED} from '../auth';

export const createIsAuthenticatedQueryMock = (isAuthenticated = true) => {
  const queryResult = {
    data: {
      currentUser: {
        isAuthenticated,
      },
    },
  };

  const resultFunc = jest.fn().mockReturnValue(queryResult);

  const queryMock = {
    request: {
      query: GET_CURRENT_USER_IS_AUTHENTICATED,
    },
    newData: resultFunc,
  };
  return [queryMock, resultFunc];
};

export const createIsAuthenticatedQueryErrorMock = (
  error = new Error('An error has occurred.'),
) => {
  const queryMock = {
    request: {
      query: GET_CURRENT_USER_IS_AUTHENTICATED,
    },
    error,
  };
  return [queryMock];
};
