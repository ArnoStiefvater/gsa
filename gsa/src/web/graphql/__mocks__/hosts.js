/* Copyright (C) 2021 Greenbone Networks GmbH
 *
 * SPDX-License-Identifier: AGPL-3.0-or-later
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Affero General Public License
 * as published by the Free Software Foundation, either version 3
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */

import {deepFreeze, createGenericQueryMock} from 'web/utils/testing';
import {
  CREATE_HOST,
  DELETE_HOST,
  DELETE_HOSTS_BY_FILTER,
  DELETE_HOSTS_BY_IDS,
  EXPORT_HOSTS_BY_FILTER,
  EXPORT_HOSTS_BY_IDS,
  GET_HOST,
  GET_HOSTS,
  MODIFY_HOST,
} from '../hosts';

export const host = deepFreeze({
  id: '12345',
  name: 'Foo',
  comment: 'bar',
  owner: {name: 'admin'},
  creationTime: '2019-06-02T12:00:22Z',
  modificationTime: '2019-06-03T11:00:22Z',
  writable: true,
  inUse: false,
  permissions: [{name: 'everything'}],
  severity: 10.0,
  details: [
    {
      name: 'best_os_cpe',
      value: 'cpe:/o:linux:kernel',
      source: {
        type: 'Report',
        description: null,
      },
      extra: null,
    },
    {
      name: 'best_os_txt',
      value: 'Linux/Unix',
      source: {
        type: 'Report',
        description: null,
      },
      extra: null,
    },
    {
      name: 'traceroute',
      value: '123.456.789.10,123.456.789.11',
      source: {
        type: 'Report',
        description: null,
      },
      extra: null,
    },
  ],
  routes: [
    {
      hosts: [
        {
          id: '10',
          ip: '123.456.789.10',
          distance: null,
          sameSource: null,
        },
        {
          id: '01',
          ip: '123.456.789.11',
          distance: null,
          sameSource: null,
        },
      ],
    },
  ],
  identifiers: [
    {
      id: '5678',
      name: 'hostname',
      value: 'foo',
      creationTime: '2019-06-02T12:00:22Z',
      modificationTime: '2019-06-03T11:00:22Z',
      sourceId: '910',
      sourceType: 'Report Host Detail',
      sourceData: '1.2.3.4.5',
      sourceName: null,
      sourceDeleted: null,
      osId: null,
      osTitle: null,
    },
    {
      id: '1112',
      name: 'ip',
      value: '123.456.789.10',
      creationTime: '2019-06-02T12:00:22Z',
      modificationTime: '2019-06-03T11:00:22Z',
      sourceId: '910',
      sourceType: 'Report Host Detail',
      sourceData: '1.2.3.4.5',
      sourceName: null,
      sourceDeleted: null,
      osId: null,
      osTitle: null,
    },
    {
      id: '1314',
      name: 'OS',
      value: 'cpe:/o:linux:kernel',
      creationTime: '2019-06-02T12:00:22Z',
      modificationTime: '2019-06-03T11:00:22Z',
      sourceId: '910',
      sourceType: 'Report Host Detail',
      sourceData: '1.2.3.4.5',
      sourceName: null,
      sourceDeleted: null,
      osId: '1314',
      osTitle: 'TestOs',
    },
  ],
});

export const hostWithoutPermission = deepFreeze({
  id: '12345',
  name: 'Foo',
  comment: 'bar',
  owner: {name: 'admin'},
  creationTime: '2019-06-02T12:00:22Z',
  modificationTime: '2019-06-03T11:00:22Z',
  writable: true,
  inUse: false,
  permissions: [{name: 'get_assets'}],
  severity: 10.0,
});

const mockHosts = {
  edges: [
    {
      node: host,
    },
  ],
  counts: {
    total: 1,
    filtered: 1,
    offset: 0,
    limit: 10,
    length: 1,
  },
  pageInfo: {
    hasNextPage: false,
    hasPreviousPage: false,
    startCursor: 'c2NoZWR1bGU6MA==',
    endCursor: 'c2NoZWR1bGU6MA==',
    lastPageCursor: 'c2NoZWR1bGU6MA==',
  },
};

export const createGetHostsQueryMock = (variables = {}) => {
  const queryResult = {
    data: {
      hosts: mockHosts,
    },
  };

  const resultFunc = jest.fn().mockReturnValue(queryResult);

  const queryMock = {
    request: {
      query: GET_HOSTS,
      variables,
    },
    newData: resultFunc,
  };
  return [queryMock, resultFunc];
};

export const createGetHostQueryMock = (id = '12345', result = host) =>
  createGenericQueryMock(GET_HOST, {host: result}, {id});

const exportHostsByIdsResult = {
  exportHostsByIds: {
    exportedEntities: '<get_assets_response status="200" status_text="OK" />',
  },
};

export const createExportHostsByIdsQueryMock = (hostIds = ['234']) =>
  createGenericQueryMock(EXPORT_HOSTS_BY_IDS, exportHostsByIdsResult, {
    ids: hostIds,
  });

const exportHostsByFilterResult = {
  exportHostsByFilter: {
    exportedEntities: '<get_assets_response status="200" status_text="OK" />',
  },
};

export const createExportHostsByFilterQueryMock = (filterString = 'foo') => {
  return createGenericQueryMock(
    EXPORT_HOSTS_BY_FILTER,
    exportHostsByFilterResult,
    {filterString},
  );
};

const bulkDeleteByIdsResult = {
  deleteHostsByIds: {
    ok: true,
  },
};

export const createDeleteHostsByIdsQueryMock = (hostIds = ['234']) =>
  createGenericQueryMock(DELETE_HOSTS_BY_IDS, bulkDeleteByIdsResult, {
    ids: hostIds,
  });

const bulkDeleteByFilterResult = {
  deleteHostsByFilter: {
    ok: true,
  },
};

export const createDeleteHostsByFilterQueryMock = (filterString = 'foo') =>
  createGenericQueryMock(DELETE_HOSTS_BY_FILTER, bulkDeleteByFilterResult, {
    filterString,
  });

const deleteHostResult = {
  deleteHostByIds: {
    ok: true,
    status: 200,
  },
};

export const createDeleteHostQueryMock = (hostId = 'foo') =>
  createGenericQueryMock(DELETE_HOST, deleteHostResult, {
    id: hostId,
  });

export const createHostInput = {
  name: 'Foo',
  comment: 'bar',
};

const createHostResult = {
  createHost: {
    id: '12345',
    status: 200,
  },
};

export const createCreateHostQueryMock = () =>
  createGenericQueryMock(CREATE_HOST, createHostResult, {
    input: createHostInput,
  });

export const modifyHostInput = {
  id: '12345',
  comment: 'bar',
};

const modifyHostResult = {
  modifyHost: {
    ok: true,
  },
};

export const createModifyHostQueryMock = () =>
  createGenericQueryMock(MODIFY_HOST, modifyHostResult, {
    input: modifyHostInput,
  });
