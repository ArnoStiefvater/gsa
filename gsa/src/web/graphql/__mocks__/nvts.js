/* Copyright (C) 2020-2021 Greenbone Networks GmbH
 *
 * SPDX-License-Identifier: AGPL-3.0-or-later
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Affero General Public License
 * as published by the Free Software Foundation, either version 3
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */

import {deepFreeze, createGenericQueryMock} from 'web/utils/testing';

import {
  GET_NVTS,
  GET_NVT,
  EXPORT_NVTS_BY_IDS,
  EXPORT_NVTS_BY_FILTER,
} from '../nvts';

export const nvtEntity = deepFreeze({
  id: '12345',
  name: '12345',
  comment: '',
  writable: 0,
  owner: null,
  inUse: 0,
  creationTime: '2019-06-24T11:55:30Z',
  modificationTime: '2019-06-24T10:12:27Z',
  updateTime: '2020-10-30T11:44:00.000+0000',
  permissions: null,
  userTags: null,
  category: 3,
  family: 'bar',
  cvssBase: 4.9,
  qod: {
    value: 80,
    type: 'remote_banner',
  },
  severities: {
    score: 49,
    severitiesList: [
      {
        date: '2020-10-30T11:44:00.000+0000',
        origin: null,
        score: 49,
        type: '',
        vector: '',
      },
    ],
  },
  refs: {
    warning: null,
    refList: [
      {type: 'cve', id: 'CVE-2020-1234'},
      {type: 'cve', id: 'CVE-2020-5678'},
    ],
  },
  tags: {
    cvssBaseVector: 'AV:N/AC:M/Au:S/C:P/I:N/A:P',
    summary: 'This is a description',
    solutionType: 'VendorFix',
    insight: 'Foo',
    impact: 'Bar',
    vuldetect: 'Baz',
    affected: 'foo',
  },
  preferencesCount: -1,
  preferences: {
    nvt: null,
    hrName: 'meh',
    name: 'meh',
    id: 1,
    type: 'ratio',
    alt: ['moo'],
    default: 'miau',
    value: 'miau',
  },
  timeout: null,
  defaultTimeout: null,
  solution: {
    type: 'VendorFix',
    description: 'This is a description',
    method: null,
  },
});

const mockNvts = {
  edges: [
    {
      node: nvtEntity,
    },
  ],
  counts: {
    total: 1,
    filtered: 1,
    offset: 0,
    limit: 10,
    length: 1,
  },
  pageInfo: {
    hasNextPage: false,
    hasPreviousPage: false,
    startCursor: 'c2NoZWR1bGU6MA==',
    endCursor: 'c2NoZWR1bGU6MA==',
    lastPageCursor: 'c2NoZWR1bGU6MA==',
  },
};

export const createGetNvtQueryMock = (id = '12345', result = nvtEntity) =>
  createGenericQueryMock(GET_NVT, {nvt: result}, {id});

export const createGetNvtsQueryMock = (variables = {}) =>
  createGenericQueryMock(GET_NVTS, {nvts: mockNvts}, variables);

const exportNvtsByIds = {
  exportNvtsByIds: {
    exportedEntities:
      '<get_info_list_response status="200" status_text="OK" />',
  },
};

export const createExportNvtsByIdsQueryMock = (nvtIds = ['12345']) =>
  createGenericQueryMock(EXPORT_NVTS_BY_IDS, exportNvtsByIds, {ids: nvtIds});

const exportNvtsByFilter = {
  exportNvtsByFilter: {
    exportedEntities:
      '<get_info_list_response status="200" status_text="OK" />',
  },
};

export const createExportNvtsByFilterQueryMock = (filterString = 'foo') =>
  createGenericQueryMock(EXPORT_NVTS_BY_FILTER, exportNvtsByFilter, {
    filterString,
  });
