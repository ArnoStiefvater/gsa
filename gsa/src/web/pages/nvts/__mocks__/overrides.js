/* Copyright (C) 2021 Greenbone Networks GmbH
 *
 * SPDX-License-Identifier: AGPL-3.0-or-later
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Affero General Public License
 * as published by the Free Software Foundation, either version 3
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import {deepFreeze} from 'web/utils/testing';

import {GET_OVERRIDES} from 'web/graphql/overrides';

const nvtOverride = deepFreeze({
  id: '456',
  active: true,
  creationTime: '2021-01-14T05:35:57Z',
  endTime: '2021-03-13T11:35:20Z',
  hosts: ['127.0.0.1', '127.0.0.2'],
  inUse: false,
  modificationTime: '2021-01-14T06:20:57Z',
  newSeverity: 10.0,
  nvt: {
    id: '12345',
    name: 'foo',
  },
  owner: 'admin',
  result: {
    id: '1337',
    name: 'result name',
  },
  permissions: [{name: 'Everything'}],
  port: '666/tcp',
  severity: 0.0,
  task: null,
  text: 'test_override_1',
  userTags: {
    count: 1,
    tags: [
      {
        id: '123',
        name: 'override:unnamed',
        value: null,
        comment: null,
      },
    ],
  },
  writable: true,
});

const nvtOverride2 = deepFreeze({
  id: '123',
  hosts: ['127.0.0.1', '127.0.0.2'],
  nvt: {
    id: '12345',
    name: 'foo nvt',
  },
  permissions: [{name: 'Everything'}],
  port: '666/tcp',
  task: {
    name: 'task x',
  },
  text: 'override text',
});

const mockNvtOverrides = {
  edges: [
    {
      node: nvtOverride,
    },
    {
      node: nvtOverride2,
    },
  ],
  counts: {
    total: 1,
    filtered: 1,
    offset: 0,
    limit: 10,
    length: 1,
  },
  pageInfo: {
    hasNextPage: false,
    hasPreviousPage: false,
    startCursor: 'override:0',
    endCursor: 'override:1',
    lastPageCursor: 'override:2',
  },
};

export const createGetOverridesQueryMock = (variables = {}) => {
  const queryResult = {
    data: {
      overrides: mockNvtOverrides,
    },
  };

  const resultFunc = jest.fn().mockReturnValue(queryResult);

  const queryMock = {
    request: {
      query: GET_OVERRIDES,
      variables,
    },
    newData: resultFunc,
  };
  return [queryMock, resultFunc];
};
