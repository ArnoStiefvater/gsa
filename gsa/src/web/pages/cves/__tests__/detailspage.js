/* Copyright (C) 2019-2021 Greenbone Networks GmbH
 *
 * SPDX-License-Identifier: AGPL-3.0-or-later
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Affero General Public License
 * as published by the Free Software Foundation, either version 3
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import React from 'react';

import {setLocale} from 'gmp/locale/lang';

import {isDefined} from 'gmp/utils/identity';

import Capabilities from 'gmp/capabilities/capabilities';

import Cve from 'gmp/models/cve';

import {entityLoadingActions} from 'web/store/entities/cves';
import {
  createExportCvesByIdsQueryMock,
  createGetCveQueryMock,
  cveEntity,
} from 'web/graphql/__mocks__/cves';
import {setTimezone, setUsername} from 'web/store/usersettings/actions';

import {createRenewSessionQueryMock} from 'web/graphql/__mocks__/session';
import {
  createGetPermissionsQueryMock,
  noPermissions,
} from 'web/graphql/__mocks__/permissions';

import {rendererWith, wait} from 'web/utils/testing';

import CvePage from '../detailspage';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({
    id: 'CVE-314',
  }),
}));

if (!isDefined(window.URL)) {
  window.URL = {};
}
window.URL.createObjectURL = jest.fn();

setLocale('en');

const entity_v2 = Cve.fromElement({
  _id: 'CVE-2020-9997',
  owner: {
    name: '',
  },
  name: 'CVE-2020-9997',
  comment: '',
  creation_time: '2020-10-22T19:15:00Z',
  modification_time: '2020-10-26T20:27:00Z',
  writable: 0,
  in_use: 0,
  permissions: '',
  update_time: '2020-10-30T11:44:00.000+0000',
  cve: {
    score: 55,
    cvss_vector: 'CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:U/C:H/I:N/A:N',
    description:
      'An information disclosure issue was addressed with improved state management. This issue is fixed in macOS Catalina 10.15.6, watchOS 6.2.8. A malicious application may disclose restricted memory.',
    products: 'cpe:/o:apple:mac_os_x:10.15.5 cpe:/o:apple:watchos:6.2.8',
    nvts: '',
    cert: {
      cert_ref: {
        _type: 'CERT-Bund',
        name: 'CB-K20/0730',
        title: 'Apple macOS: Mehrere Schwachstellen',
      },
    },
    raw_data: {
      entry: {
        _id: 'CVE-2020-9997',
        'vulnerable-software-list': {
          product: [
            'cpe:/o:apple:mac_os_x:10.15.5',
            'cpe:/o:apple:watchos:6.2.8',
          ],
        },
        'cve-id': 'CVE-2020-9997',
        'published-datetime': '2020-10-22T19:15:00+00:00',
        'last-modified-datetime': '2020-10-26T20:27:00+00:00',
        cvss: {
          base_metrics: {
            'integrity-impact': 'NONE',
            'access-complexity': 'MEDIUM',
            'availability-impact': 'NONE',
            'vector-string': 'AV:N/AC:M/Au:N/C:P/I:N/A:N',
            'confidentiality-impact': 'PARTIAL',
            'access-vector': 'NETWORK',
            authentication: 'NONE',
            score: 4.3,
            source: 'http://nvd.nist.gov',
            'generated-on-datetime': '2020-10-26T20:27:00+00:00',
          },
        },
        cvss3: {
          base_metrics: {
            'availability-impact': 'NONE',
            'attack-complexity': 'LOW',
            'vector-string': 'CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:U/C:H/I:N/A:N',
            'integrity-impact': 'NONE',
            'user-interaction': 'REQUIRED',
            'base-score': 5.5,
            'attack-vector': 'LOCAL',
            'privileges-required': 'NONE',
            scope: 'UNCHANGED',
            'confidentiality-impact': 'HIGH',
            'base-severity': 'MEDIUM',
          },
          'generated-on-datetime': '2020-10-26T20:27:00+00:00',
        },
        cwe: {
          _id: 'NVD-CWE-noinfo',
        },
        references: [
          {
            source: 'MISC',
            reference: {
              __text: 'https://support.apple.com/kb/HT211289',
              _href: 'https://support.apple.com/kb/HT211289',
            },
          },
          {
            source: 'MISC',
            reference: {
              __text: 'https://support.apple.com/kb/HT211291',
              _href: 'https://support.apple.com/kb/HT211291',
            },
          },
        ],
        summary:
          'An information disclosure issue was addressed with improved state management. This issue is fixed in macOS Catalina 10.15.6, watchOS 6.2.8. A malicious application may disclose restricted memory.',
      },
    },
  },
});
const cve = Cve.fromObject(cveEntity);

const caps = new Capabilities(['everything']);
const wrongCaps = new Capabilities(['get_config']);

const entityType = 'cve';
const reloadInterval = -1;
const manualUrl = 'test/';

let currentSettings;
let getPermissions;
let renewSession;

beforeEach(() => {
  currentSettings = jest.fn().mockResolvedValue({
    foo: 'bar',
  });

  renewSession = jest.fn().mockResolvedValue({
    foo: 'bar',
  });
});

describe('CVE Detailspage tests', () => {
  test('should render full Detailspage', async () => {
    const getCve = jest.fn().mockResolvedValue({
      data: cve,
    });

    const gmp = {
      [entityType]: {
        get: getCve,
      },
      permissions: {
        get: getPermissions,
      },
      settings: {manualUrl, reloadInterval},
      user: {
        currentSettings,
      },
    };

    const [mock, resultFunc] = createGetCveQueryMock();
    const [renewSessionQueryMock] = createRenewSessionQueryMock();

    const [permissionQueryMock] = createGetPermissionsQueryMock({
      filterString: 'resource_uuid=314 first=1 rows=-1',
    });
    const {render, store} = rendererWith({
      capabilities: caps,
      gmp,
      router: true,
      store: true,
      queryMocks: [mock, renewSessionQueryMock],
    });

    store.dispatch(setTimezone('UTC'));
    store.dispatch(setUsername('admin'));

    // store.dispatch(entityLoadingActions.success('CVE-314', entity_v2));

    const {baseElement, element, getAllByTestId} = render(
      <CvePage id="CVE-314" />,
    );

    await wait();

    expect(resultFunc).toHaveBeenCalled();

    const links = baseElement.querySelectorAll('a');
    const icons = getAllByTestId('svg-icon');

    expect(icons[0]).toHaveAttribute('title', 'Help: CVEs');
    expect(links[0]).toHaveAttribute(
      'href',
      'test/en/managing-secinfo.html#cve',
    );

    expect(icons[1]).toHaveAttribute('title', 'CVE List');
    expect(links[1]).toHaveAttribute('href', '/cves');

    expect(icons[2]).toHaveAttribute('title', 'Export CVE');

    expect(baseElement).toHaveTextContent('CVE: foo');

    expect(baseElement).toHaveTextContent('CVE-314');
    expect(baseElement).toHaveTextContent(
      'Published:Mon, Aug 17, 2020 12:18 PM UTC',
    );
    expect(baseElement).toHaveTextContent(
      'Modified:Tue, Sep 29, 2020 12:16 PM UTC',
    );
    expect(baseElement).toHaveTextContent(
      'Last updated:Tue, Sep 29, 2020 12:16 PM UTC',
    );

    expect(baseElement).toHaveTextContent('Attack VectorLOCAL');
    expect(baseElement).toHaveTextContent('Attack ComplexityLOW');
    expect(baseElement).toHaveTextContent('Privileges RequiredNONE');
    expect(baseElement).toHaveTextContent('User InteractionREQUIRED');
    expect(baseElement).toHaveTextContent('ScopeUNCHANGED');
    expect(baseElement).toHaveTextContent('Confidentiality ImpactHIGH');
    expect(baseElement).toHaveTextContent('Integrity ImpactNONE');
    expect(baseElement).toHaveTextContent('Availability ImpactNONE');
    const progressBars = getAllByTestId('progressbar-box');
    expect(progressBars[0]).toHaveAttribute('title', 'Medium');
    expect(progressBars[0]).toHaveTextContent('5.5 (Medium)');
    expect(baseElement).toHaveTextContent('References');
    expect(baseElement).toHaveTextContent(
      'MISChttps://support.apple.com/kb/HT211289',
    );
    expect(element).toHaveTextContent(
      'MISChttps://support.apple.com/kb/HT211291',
    );
    expect(baseElement).toHaveTextContent(
      'CERT Advisories referencing this CVE',
    );
    expect(baseElement).toHaveTextContent('CB-K20/0730');
    expect(baseElement).toHaveTextContent(
      'Apple macOS: Mehrere Schwachstellen',
    );
    expect(baseElement).toHaveTextContent('Vulnerable Products');
    expect(baseElement).toHaveTextContent('cpe:/o:apple:mac_os_x:10.15.5');
    expect(baseElement).toHaveTextContent('cpe:/o:apple:watchos:6.2.8');
  });
});
