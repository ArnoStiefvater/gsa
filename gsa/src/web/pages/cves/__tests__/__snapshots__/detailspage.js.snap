// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CVEs ToolBarIcons tests should render 1`] = `
.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: -10px;
}

.c1 > * {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c1 > * {
  margin-left: 10px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: -5px;
}

.c2 > * {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c2 > * {
  margin-left: 5px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c5 {
  cursor: pointer;
}

.c3 {
  height: 16px;
  width: 16px;
  line-height: 16px;
}

.c3 * {
  height: inherit;
  width: inherit;
}

@media print {
  .c5 {
    display: none;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
    margin="10px"
  >
    <div
      class="c0"
    >
      <div
        class="c2"
        margin="5px"
      >
        <a
          href="test/en/managing-secinfo.html#cve"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span
            class="c3"
            data-testid="svg-icon"
            title="Help: CVEs"
          >
            <svg
              title="Help: CVEs"
            >
              help.svg
            </svg>
          </span>
        </a>
        <a
          class="c4"
          href="/cves"
        >
          <span
            class="c3"
            data-testid="svg-icon"
            title="CVE List"
          >
            <svg
              title="CVE List"
            >
              list.svg
            </svg>
          </span>
        </a>
      </div>
    </div>
    <span
      class="c5 c3"
      data-testid="svg-icon"
      title="Export CVE"
    >
      <svg
        title="Export CVE"
      >
        export.svg
      </svg>
    </span>
  </div>
</div>
`;
